src/webgpu/shader/validation/parse/for.spec.ts

```diff
@@ -0,0 +1,50 @@
+export const description = `Validation parser tests for 'for' statements`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../common/util/data_tables.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kTests = {
+  break: { wgsl: `for (;;) { break; }`, pass: true },
+  init_var: { wgsl: `for (var i = 1;;) { break; }`, pass: true },
+  init_var_type: { wgsl: `for (var i : i32 = 1;;) { break; }`, pass: true },
+  init_var_function: { wgsl: `for (var<function> i = 1;;) { break; }`, pass: true },
+  init_var_function_type: { wgsl: `for (var<function> i : i32 = 1;;) { break; }`, pass: true },
+  init_let: { wgsl: `for (let i = 1;;) { break; }`, pass: true },
+  init_let_type: { wgsl: `for (let i : u32 = 1;;) { break; }`, pass: true },
+  init_const: { wgsl: `for (const i = 1;;) { break; }`, pass: true },
+  init_const_type: { wgsl: `for (const i : f32 = 1;;) { break; }`, pass: true },
+  init_call: { wgsl: `for (x();;) { break; }`, pass: true },
+  cond: { wgsl: `for (;true;) { break; }`, pass: true },
+  cont: { wgsl: `for (;;v++) { break; }`, pass: true },
+
+  init_cond: { wgsl: `for (var i = 1; i < 5;) {}`, pass: true },
+  cond_cont: { wgsl: `for (;v < 5; v++) {}`, pass: true },
+  init_cond_cont: { wgsl: `for (var i = 0; i < 5; i++) {}`, pass: true },
+  init_shadow: { wgsl: `for (var f = 0; f < 5; f++) {}`, pass: true },
+
+  no_semicolon: { wgsl: `for () { break; }`, pass: false },
+  one_semicolon: { wgsl: `for (;) { break; }`, pass: false },
+  no_paren: { wgsl: `for ;; { break; }`, pass: false },
+  empty: { wgsl: `for (;;) {}`, pass: false },
```

Comment:
Done. I agree that this is confusing. I'm not keen on the structure here - there's a whole bunch of 'parse' tests which are nothing to do with parsing (at least not for Tint's implementation). IMO it would make sense to pull all this stuff out of here and into the other `src/webgpu/shader/validation/...` files / directories, and keep all the tests together for each of the respective grammar elements. I didn't want to shuffle code with this PR.